src/main/java/kitchenpos/menu/domain/MenuProducts.java

```diff
@@ -0,0 +1,45 @@
+package kitchenpos.menu.domain;
+
+import javax.persistence.CascadeType;
+import javax.persistence.OneToMany;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class MenuProducts {
+
+    @OneToMany(mappedBy = "menu", cascade = CascadeType.ALL, orphanRemoval = true)
+    private List<MenuProduct> menuProducts = new ArrayList<>();
+
+    protected MenuProducts() {
+    }
+
+    protected MenuProducts(List<MenuProduct> menuProducts) {
+        this.menuProducts = menuProducts;
+    }
+
+    public static MenuProducts of(List<MenuProduct> menuProducts) {
+        return new MenuProducts(menuProducts);
+    }
+
+    public List<MenuProduct> getMenuProducts() {
+        return Collections.unmodifiableList(menuProducts);
+    }
+
+    public BigDecimal getSumOfMenuPrice() {
+        BigDecimal sum = BigDecimal.ZERO;
+        for (final MenuProduct menuProduct : menuProducts) {
+            sum = sum.add(menuProduct.getProduct().getPrice().multiply(BigDecimal.valueOf(menuProduct.getQuantity())));
```

Comment:
```suggestion
            sum = sum.add(menuProduct.multiply(menuProduct.getQuantity()));
```
getter 체이닝이 너무 많이 걸려서 가독성이 조금 낮아진거 같아요! menuProduct 객체 안으로 해당로직을 가져가면어떨까요?

Reply:
menuProduct 안에서 로직을 옮기니 훨씬 깔끔해진거같습니다 감사합니다! :)